amanuensis/resources/associated_user/__init__.py

```diff
@@ -23,44 +24,82 @@ def add_associated_users(session, users, role=None):
     input_users = [defaultdict(lambda: None, user) for user in users]
     for input_user in input_users:
 
-    
+        email = input_user["email"] if "email" in input_user else None
+        id = input_user["id"] if "id" in input_user else None
         
         if "project_id" not in input_user:
             raise UserError("The project_id is required to add user to project")
+        
+        get_projects(session, id=input_user["project_id"], throw_not_found=True, many=False)
 
-        if "email" in input_user:
-            fence_user = fence_get_users(usernames=[input_user["email"]])["users"]
+        if email:
+            fence_user = fence_get_users(usernames=[email])["users"]
 
             if fence_user:
-                if input_user["id"] and input_user["id"] != fence_user[0]["id"]:
-                    raise UserError(f"The user id {input_user['id']} does not exist in the commons")
+                if id and id != fence_user[0]["id"]:
+                    raise UserError(f"The user id {id} does not exist in the commons")
                 else:
-                    input_user["id"] = fence_user[0]["id"]
+                    id = fence_user[0]["id"]
+            
+            elif id:
+                raise UserError(f"The user id {id} does not match the email {email} in the commons")
```

Comment:
If there is no email but there is an id should't it try to look by id instead of saying `raise UserError(f"The user id {id} does not match the email {email} in the commons")`?